src/components/map/Map.tsx

```diff
@@ -157,141 +179,256 @@ export const Map = ({
-                        : isActiveMarker
-                        ? colors.accent
-                        : undefined
-                    }
-                  />
-                  <View
-                    style={[
-                      styles.mapIconOnLocationMarker,
-                      isActiveMarker ? styles.mapIconOnLocationMarkerActive : undefined,
-                      !!marker.iconBackgroundColor && {
-                        backgroundColor: isActiveMarker
-                          ? marker.iconColor
-                          : marker.iconBackgroundColor
-                      }
-                    ]}
-                  >
+              <>
+                {calloutTextEnabled && isActiveMarker && (
+                  <View style={styles.callout}>
+                    <View style={styles.calloutTriangle} />
+                    {!!serviceName && (
+                      <RegularText smallest center>
+                        {serviceName}
+                      </RegularText>
+                    )}
+
+                    {!!title && (
+                      <RegularText smallest center>
+                        {title}
+                      </RegularText>
+                    )}
+                  </View>
+                )}
+
+                <View style={styles.markerContainer}>
+                  {!!marker.iconName &&
+                  marker.iconName != 'ownLocation' &&
+                  marker.iconName != 'location' ? (
+                    <>
+                      <MapIcon
+                        iconColor={
+                          marker.iconBackgroundColor
+                            ? isActiveMarker
+                              ? marker.iconColor
+                              : marker.iconBackgroundColor
+                            : isActiveMarker
+                            ? colors.accent
+                            : undefined
+                        }
+                      />
+                      <View
+                        style={[
+                          styles.mapIconOnLocationMarker,
+                          isActiveMarker ? styles.mapIconOnLocationMarkerActive : undefined,
+                          !!marker.iconBackgroundColor && {
+                            backgroundColor: isActiveMarker
+                              ? marker.iconColor
+                              : marker.iconBackgroundColor
+                          }
+                        ]}
+                      >
+                        <MapIcon
+                          iconColor={
+                            marker.iconColor
+                              ? isActiveMarker
+                                ? colors.surface
+                                : marker.iconColor
+                              : colors.surface
+                          }
+                          iconName={marker.iconName}
+                          iconSize={MARKER_ICON_SIZE / 3.25}
+                        />
```

Comment:
I have noticed that if there is a pin after the map is loaded, the map updates itself late or we have to move it to update the map. I haven't found a solution so far, but I found where the problem is coming from.

The problem is caused by the custom marker we developed. In Map.tsx, if we make 332 to 397 of the code a comment line, the map loads perfectly.  Otherwise the map first shows the whole world and places the pin in the center of the world. If we wait for a certain time or move the map we can see the exact location.

|beginning|after moving the map|
|--|--|
![Simulator Screenshot - iPhone 15 Pro Max - 2024-07-22 at 17 30 40](https://github.com/user-attachments/assets/96121796-c695-46ae-a261-e097796c9a32)|![Simulator Screenshot - iPhone 15 Pro Max - 2024-07-22 at 17 30 45](https://github.com/user-attachments/assets/1df3e5b1-1073-43c4-8ed1-4f2873fbd3ff)